### ISSUE TITLE
Cannot login

### ISSUE THREAD

**Issue by AlexSanzharevskyy** [chalasr/RCHJWTUserBundle#8]
Hi there!

I've setup Symfony 3.1 from the scratch, and after - your bundle. I went through all the steps. But i cant register user (error": "Request parameter email must be set") or even login with existed (code": 401, "message": "Bad credentials")

**Comment by chalasr** [comment]
Hi @AlexSanzharevskyy,

Thank you for using this bundle and reporting this issue.
I'll give it a try tomorrow and keep you informed here.

**Comment by cve** [comment]
I think that this is problem with Content-Type header, if I set Content-Type:application/x-www-form-urlencoded this works, but with standard application/json the request in SecurityController is not decoded

**Comment by cve** [comment]
I made a PR #10

**Comment by AlexSanzharevskyy** [comment]
I've setup all from scratch (FOSUser, JWT, FOSRest, NelmioCORS). And now it works. I think there were 2 problems - the first one you mentioned, the second - with cors.

**Comment by cve** [comment]
I think, FOSRest has body listener which converts incoming json request automtically, I'm using RCHJWTUserBundle with Api Platform, which does not decode every request with listener I think, but I can be wrong

**Comment by chalasr** [comment]
@cve Good catch, merged #10.
@AlexSanzharevskyy Do you think we can close this issue?

**Comment by AlexSanzharevskyy** [comment]
Yes, please. I'll write my configs, maybe it will help someone)

**config.yml**

#FOSRest
`fos_rest:
    body_listener: true
    format_listener:  true
    param_fetcher_listener: true
    view:
        view_response_listener: 'force'
        exception_wrapper_handler:  null
        formats:
            jsonp: true
            json: true
            xml: false
            rss: false
        mime_types:
            json: ['application/json', 'application/x-json']
            jpg: 'image/jpeg'
            png: 'image/png'
        jsonp_handler: ~
    routing_loader:
        default_format:  json
        include_format:  false
    format_listener:
        rules:
            - { path: ^/, priorities: [ json, jsonp ], fallback_format: ~, prefer_extension: true }
    exception:
        enabled: true`

# JMS Serializer
`jms_serializer:
    metadata:
        directories:
            FOSUB:
                namespace_prefix: FOS\UserBundle
                path: %kernel.root_dir%/serializer/FOSUB`

#Nelmio CORS
`nelmio_cors:
    defaults:
        allow_credentials: false
        allow_origin: []
        allow_headers: []
        allow_methods: []
        expose_headers: []
        max_age: 0
        hosts: []
        origin_regex: false

    paths:
        '^/':
            allow_origin: ['*']
            allow_headers: ['X-Custom-Auth', 'Content-Type', 'X-Requested-With', 'accept', 'Origin', 'Access-Control-Request-Method', 'Access-Control-Request-Headers', 'Authorization', 'Cache-Control', 'if-modified-since', 'pragma']
            allow_methods: ['POST', 'PUT', 'GET', 'DELETE']
            max_age: 3600`

**security.yml**
`security:
    encoders:
        FOS\UserBundle\Model\UserInterface: bcrypt

    role_hierarchy:
        ROLE_ADMIN:       ROLE_USER
        ROLE_SUPER_ADMIN: ROLE_ADMIN

    providers:
        fos_userbundle:
            id: fos_user.user_provider.username_email

    firewalls:
        login:
            pattern:  ^/api/login
            stateless: true
            anonymous: true
            form_login:
                provider: fos_userbundle
                check_path: /api/login
                success_handler: lexik_jwt_authentication.handler.authentication_success
                failure_handler: lexik_jwt_authentication.handler.authentication_failure
                require_previous_session: false
                username_parameter: username
                password_parameter: password

        api:
            pattern:   ^/api
            stateless: true
            lexik_jwt:
                authorization_header:
                    enabled: true
                    prefix:  Bearer
                    name:    Authorization
                cookie:
                    enabled: false
                    name:    BEARER
                query_parameter:
                    enabled: false
                    name:    bearer
                throw_exceptions: false
                create_entry_point: true
                authentication_provider: lexik_jwt_authentication.security.authentication.provider
                authentication_listener: lexik_jwt_authentication.security.authentication.listener

    access_control:
        - { path: ^/api/doc,        roles: IS_AUTHENTICATED_ANONYMOUSLY }
        - { path: ^/api/register,   roles: IS_AUTHENTICATED_ANONYMOUSLY }
        - { path: ^/api/login,      roles: IS_AUTHENTICATED_ANONYMOUSLY }
        - { path: ^/api,            roles: IS_AUTHENTICATED_FULLY }`

**Comment by rienk-git** [comment]
Just a comment on this.

After  having this exact same problem but with a very different cause, which was my own mistake (as is most of the time), I post this because someone else might have similar problems, finds this page, and made a similar mistake as I did.

I had to reinstall my devbox, so I pulled my project from github and got this error (401 on a call but able to login and register).

The problem was that I forgot to include:
``` 
RewriteEngine On
RewriteCond %{HTTP:Authorization} ^(.*)
RewriteRule .* - [e=HTTP_AUTHORIZATION:%1]
```

In my vhost file (which is offcourse in the lexikjwt docs, i know) !
After inspecting the request using the symfony debugbar it was an easy fix but I made the mistake of googling first, inspect later :(.
